src/nuget/Fable.JsonConverter/Fable.JsonConverter.fs

```diff
@@ -36,15 +36,15 @@ type JsonConverter() =
         advance reader
         read 0 List.empty
 
-    let typeCache = Dictionary<Type, bool>()
+    let typeCache = ConcurrentDictionary<_, _>()
     override x.CanConvert(t) =
             match typeCache.TryGetValue(t) with
             | false, _ ->
                 let result = 
                     t.Name = "FSharpOption`1"
                     || FSharpType.IsTuple t
                     || (FSharpType.IsUnion t && t.Name <> "FSharpList`1")
-                typeCache.Add(t, result)
+                typeCache.TryAdd(t, result) |> ignore
                 result
             | _, result -> result
```

Comment:
@ncave Thanks for the clarification! :+1: One question though, the purpose of `&& t.Name <> "FSharpList`1"` is to make Json.NET deal with F# lists. If the guard is omitted, the converted will try to parse them as other union types giving unexpected results. Are you sure it's safe to remove it?